contracts/protocol/v1/proxies/P1LiquidatorProxy.sol

```diff
@@ -0,0 +1,320 @@
+        emit LogLiquidatorProxyUsed(
+            msg.sender,
+            liquidatee,
+            isBuy,
+            liqAmount,
+            feeAmount
+        );
+
+        return liqAmount;
+    }
+
+    // ============ Admin Functions ============
+
+    /**
+     * @dev Allows the owner to set the insurance fund address. Emits the LogInsuranceFundSet event.
+     *
+     * @param  insuranceFund  The address to set as the insurance fund.
+     */
+    function setInsuranceFund(
+        address insuranceFund
+    )
+        external
+        onlyOwner
+    {
+        _INSURANCE_FUND_ = insuranceFund;
+        emit LogInsuranceFundSet(insuranceFund);
+    }
+
+    /**
+     * @dev Allows the owner to set the insurance fee. Emits the LogInsuranceFeeSet event.
+     *
+     * @param  insuranceFee  The new fee as a fixed-point number with 18 decimal places. Max of 1.
+     */
+    function setInsuranceFee(
+        uint256 insuranceFee
+    )
+        external
+        onlyOwner
+    {
+        require(
+            insuranceFee <= BaseMath.base().div(2),
+            "insuranceFee cannot be greater than 50%"
+        );
+        _INSURANCE_FEE_ = insuranceFee;
+        emit LogInsuranceFeeSet(insuranceFee);
+    }
+
+    // ============ Helper Functions ============
+
+    /**
+     * @dev Calculate (and verify) the maximum amount to liquidate based on the maxPosition input.
+     */
+    function _getMaxPositionDelta(
+        P1Types.Balance memory initialBalance,
+        bool isBuy,
+        SignedMath.Int memory maxPosition
+    )
+        private
+        pure
+        returns (SignedMath.Int memory)
+    {
+        SignedMath.Int memory result = maxPosition.signedSub(initialBalance.getPosition());
+
+        require(
+            result.isPositive == isBuy && result.value > 0,
+            "Cannot liquidate past maxPosition"
+        );
+
+        return result;
+    }
+
+    /**
+     * @dev Perform the liquidation by constructing the correct arguments and sending it to the
+     * perpetual.
+     */
+    function _doLiquidation(
+        I_PerpetualV1 perpetual,
+        address liquidatee,
+        SignedMath.Int memory maxPositionDelta
+    )
+        private
+    {
+        // Create accounts.
+        bool takerFirst = address(msg.sender) < liquidatee;
+        address[] memory accounts = new address[](2);
+        uint256 takerIndex = takerFirst ? 0 : 1;
+        uint256 makerIndex = takerFirst ? 1 : 0;
+        accounts[takerIndex] = msg.sender;
+        accounts[makerIndex] = liquidatee;
+
+        // Create trade args.
+        I_PerpetualV1.TradeArg[] memory trades = new I_PerpetualV1.TradeArg[](1);
+        trades[0] = I_PerpetualV1.TradeArg({
+            takerIndex: takerIndex,
+            makerIndex: makerIndex,
+            trader: _LIQUIDATION_,
+            data: abi.encode(
+                maxPositionDelta.value,
+                maxPositionDelta.isPositive,
+                false // allOrNothing
```

Comment:
Shouldn't we allow the liquidator to pass this in? Otherwise there will be no way to use this option

Reply:
Yeah there will be no way to use this option unless we introduce another liquidator proxy but I think it's fine for now